src/renderer/wg_engine/tvgWgRenderer.cpp

```diff
@@ -246,13 +247,24 @@ bool WgRenderer::clear()
 }
 
 
+void WgRenderer::releaseSurfaceTexture()
+{
+    if (surfaceTexture.texture) {
+        wgpuTextureRelease(surfaceTexture.texture);
+        surfaceTexture.texture = nullptr;
+    }
+}
+
+
 bool WgRenderer::sync()
 {
     disposeObjects();
     if (!surface) return false;
-    // get current texture
-    WGPUSurfaceTexture surfaceTexture{};
+
+    releaseSurfaceTexture();
```

Comment:
Im not sure about this approach... Surface is owner of the screen buffer textures, and we are not _crreate_ them, but _get_ them from surface

Reply:
it is required by the wgpu guidance. and due to this all the resources are certainly leaked.